packages/bot-web-ui/src/components/dashboard/bot-stopped.spec.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/root-store';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import BotStopped from './bot-stopped';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+jest.mock('@deriv/deriv-charts', () => ({
+    setSmartChartsPublicPath: jest.fn(),
+}));
+
+describe('<BotStopped />', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    const originalLocation = global.location;
+    delete global.location;
+    global.location = {
+        ...originalLocation,
+        replace: jest.fn(),
+        reload: jest.fn(),
+    };
+
+    it('renders the BotStopped component', () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
+
+        render(<BotStopped />, {
+            wrapper,
+        });
+
+        expect(screen.getByText('You’re back online')).toBeInTheDocument();
+    });
+    it('should go to reports page on click of go to reports button', async () => {
+        mock_DBot_store?.dashboard?.setWebSocketState(false);
```

Comment:
mock_DBot_store?.dashboard?.setWebSocketState(false);

So we are using this in every case, then it should be in the before all?

Reply:
done ✅